Shared/src/com/pacoapp/paco/shared/model2/SQLQuery.java

```diff
@@ -108,10 +108,18 @@ public Builder fullEventAndOutputs(Boolean fullEventAndOutputs) {
     }
 
     private Builder addDefaultValues(SQLQuery obj) {
-        // provide default sort order which is Event._Id desc
-      if (obj.getGroupBy() == null && obj.sortOrder == null) {
-         obj.sortOrder = EventBaseColumns.TABLE_NAME+"."+Constants.UNDERSCORE_ID.concat(Constants.BLANK).concat(Constants.DESC);
-       }
+      // find if there is a distinct clause
+      boolean isDistinct = false;
+      for(String s : obj.getProjection()) {
+        if (s.startsWith(Constants.DISTINCT)) {
+          isDistinct = true;
+          break;
+        }
+      }
+      // provide default sort order which is Event._Id desc
+      if (obj.getGroupBy() == null && obj.sortOrder == null && !isDistinct) {
+        obj.sortOrder = EventBaseColumns.TABLE_NAME+"."+Constants.UNDERSCORE_ID.concat(Constants.BLANK).concat(Constants.DESC);
+      }
 
       if (obj.getProjection() == null) {
```

Comment:
Will obj.getProjection() return null, or, will it return an empty list? If it can return null, the iteration on line 113 to check for a DISTINCT needs to test for null.

Reply:
yep...you are right. I missed to check that.